### PR TITLE
DynamoDB Simple Attributes

### PR DESCRIPTION
**Pull request by awood45**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

1. To make sure we include your contribution in the release notes, please make sure to add description entry for your changes in the "unreleased changes" section of the `CHANGELOG.md` file (at corresponding gem). For the description entry, please make sure it lives in one line and starts with `Feature` or `Issue` in the correct format.

2. For generated code changes, please checkout below instructions first:
  https://github.com/aws/aws-sdk-ruby/blob/master/CONTRIBUTING.md

Thank you for your contribution!

### CODE REVIEW

**Comment by awood45** [comment]
This is currently a WIP. Final resolution of #1972

**Comment by awood45** [comment]
I just had a silly error. This version should work.

File: build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/shared_example.rb [hunk 1]
```diff
@@ -26,7 +26,12 @@ def to_str_output
 
     def entry(json, ref, indent, path)
       case ref['type']
-      when 'structure' then structure(json, ref, indent, path)
+      when 'structure'
+        if ref['members'].keys == ["S", "N", "B", "SS", "NS", "BS", "M", "L", "NULL", "BOOL"] && @api['metadata']['endpointPrefix'] == 'dynamodb'
```
Comment (cjyclaire) [inline build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/shared_example.rb]: if we are using metadata for identifying dynamo, how about a value that's unlikely to change and unique? I'd prefer service name related values or service id https://github.com/aws/aws-sdk-ruby/blob/master/apis/dynamodb/2012-08-10/api-2.json#L3

Although very rarely but it did happen that endpoint prefix changed for service models

Comment (awood45) [reply build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/shared_example.rb]: I feel like this particular one is pretty stable, but I agree this is the line that feels the most hacky in the entire change. I'm going to apply a small refactor...

**Pull request merged**